src/core/border_router/routing_manager.cpp

```diff
@@ -1719,6 +1719,21 @@ void RoutingManager::DiscoveredPrefixTable::HandleRouterTimer(void)
             continue;
         }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_SKIP_NSP_PROBE_WITH_ROUTER_ON_SAME_DEVICE
```

Comment:
Does this need to have a feature flag? @superwhd , thoughts?

Reply:
I am ok removing the `OPENTHREAD_CONFIG_BORDER_ROUTING_SKIP_NSP_PROBE_WITH_ROUTER_ON_SAME_DEVICE` (and having it be the default behavior).

Added it for now so we have option of enabling/disabling this easily if we need to

Reply:
I think we may need to always have it enabled on POSIX (or we decide to solve the problem in another way). For other platforms I don't have much idea.